course7/homeworks/fred/ls/main.cc

```diff
@@ -0,0 +1,138 @@
+#include <boost/filesystem.hpp>
+#include <boost/program_options.hpp>
+
+#include <numeric>
+#include <string>
+#include <cmath>
+#include <iostream>
+#include <iomanip>
+#include <sstream>
+#include <cstdlib>
+
+std::string humanReadableByteCount(size_t bytes) {
+  const int unit = 1024;
+
+  if (bytes < unit)
+  {
+    std::stringstream ss;
+    ss << std::fixed << std::setprecision(1) << bytes << 'B';
+    return ss.str();
+  }
+
+  int exp = static_cast<int>(std::log(bytes) / std::log(unit));
+  auto pre = "KMGTPE"[exp-1];
+
+  std::stringstream ss;
+  ss << std::fixed << std::setprecision(1) << (bytes / std::pow(unit, exp))
+      << pre << 'B';
+  return ss.str();
+}
+
+size_t getDirectorySize(const boost::filesystem::path& path)
+{
+  using namespace boost::filesystem;
+
+  assert(is_directory(path));
+  boost::filesystem::recursive_directory_iterator dir_iterator(path);
```

Comment:
Inconsistent qualification: you should be either using it or not

Reply:
I didn't understand what you mean. Using what?